src/wpa-supplicant.c

```diff
@@ -622,7 +627,7 @@ ni_wpa_interface_retrieve_scan(ni_wpa_interface_t *wpa_dev, ni_wireless_scan_t *
 	if (ni_wpa_interface_expire_networks(wpa_dev, scan->interval + 1) == 0) {
 		/* Nothing pruned. If we didn't receive new scan results in the
 		 * mean time, there's nothing we need to do. */
-		if (scan->timestamp == wpa_dev->scan.timestamp)
+		if (!timercmp(&scan->timestamp, &wpa_dev->scan.timestamp, !=))
```

Comment:
Is there any reason why you'd use a double negation here?

Reply:
See "man timercmp", there are some implementation that don't work properly with '==':
```Some  systems  (but  not Linux/glibc), have a broken timercmp() implementation, in which CMP of >=, <=, and == do not work; portable applications can instead use
           !timercmp(..., <)
           !timercmp(..., >)
           !timercmp(..., !=)
```

Reply:
You are totally right, marking as resolved